samples/hello_world/sample.robot

```diff
@@ -0,0 +1,11 @@
+# SPDX-License-Identifier: Apache-2.0
+
```

Comment:
This is a hello world sample for Zephyr, not for Robot or twister. To show how Robot works with Twister etc I would have it go into its own sample, test instead of overloading the hello world sample we have.

Reply:
We believe that the name of the robot file might be misleading here. It has been named after `sample.yaml`, but in fact it's a specific test tailored for the Hello World application and therefore could probably be renamed to `test.robot` or even `hello_world.robot`. Additionally it might not be fortunate to discuss the Hello World case at all as it introduces a reimplementation (and therefore duplication) of an already-existing console-harness-based test.
To avoid confusion we should probably just remove this test completely.

The shell example Robot test on the other hand, provides an added value as it contains an interactive scenario that has not been tested automatically up to this point. It uses a newly-proposed testing framework integration to verify an already-existing sample.

Having a separate sample showcasing different aspects of writing Robot tests is an interesting idea - we’ll think about a proper scenario that could be instructive to future users. As it will take a while, we believe it would make most sense to have a separate PR adding this sample.

Reply:
> To avoid confusion we should probably just remove this test completely.

I am fine with that. 
I meant we should add anything like this into the hello_world sample to show robot/renode feature, we need to do that somewhere, so shell is fine.